src/config/field-overrides/program-indicator/AnalyticsPeriodBoundaries.js

```diff
@@ -0,0 +1,182 @@
+                style={ {position: 'relative', marginRight: '14px'} }
+                options={[
+                    {
+                        text: getTranslation('report_period_before_start'),
+                        value: 'BEFORE_START_OF_REPORTING_PERIOD'
+                    },
+                    {
+                        text: getTranslation('report_period_before_end'),
+                        value: 'BEFORE_END_OF_REPORTING_PERIOD'
+                    },
+                    {
+                        text: getTranslation('report_period_after_start'),
+                        value: 'AFTER_START_OF_REPORTING_PERIOD'
+                    },
+                    {
+                        text: getTranslation('report_period_after_end'),
+                        value: 'AFTER_END_OF_REPORTING_PERIOD'
+                    }
+                ]}
+                onChange={e => props.onChange(e, "analyticsPeriodBoundaryType")}
+                value={props.analyticsPeriodBoundaryType}
+            />
+
+            <TF type="number"
+                style={ { flex: '0 1 auto' } }
+                labelText={getTranslation('period_number_offset')}
+                value={props.offsetNumberOfPeriods}
+                onChange={e => props.onChange(e, "offsetNumberOfPeriods")}
+            />
+
+            <PTDD
+                labelText={getTranslation('period_offset')}
+                onChange={e => props.onChange(e, "offsetPeriodType")}
+                style={ { marginRight: '14px' } }
+                value={props.offsetPeriodType}
+            />
+
+            <div style={{ flex: '0 1 auto' }}>
+                <RaisedButton
+                    label={getTranslation('remove_singular')}
+                    onClick={e => props.onClick(e)}/>
+            </div>
+        </div>
+    );
+}
+
+function updateSubField(element, model, onChange, idx, e, fieldName) {
+    const changeToVal = e.target.value;
+
+    const list = model.analyticsPeriodBoundaries;
+    const newEl = Object.assign({}, element, {[fieldName]: changeToVal});
+
+    list[idx] = newEl;
+
+    return onChange({ target: { value: list}})
+}
+
+function removeSubField(model, onChange, idx, e) {
+    const list = model.analyticsPeriodBoundaries;
+    const newList = list.filter((_, i) => i !== idx);
+
+    return onChange({ target: { value: newList }});
+}
+
+function addSubField(model, onChange) {
+    const list = model.analyticsPeriodBoundaries;
+    const newVal = {
+        analyticsPeriodBoundaryType: '',
+        boundaryTarget: '',
+        offsetNumberOfPeriods: 0,
+        offsetPeriodType: ''
+    };
+
+    return onChange({ target: {
+        value: list.concat([newVal])
+    }});
+}
+
+function AnalyticsPeriodBoundaryList ({ d2, model, onChange }) {
+    const getTranslation = d2.i18n.getTranslation.bind(d2.i18n);
+
+    const boundaries = model.analyticsPeriodBoundaries.map((props, i) => (
```

Comment:
This is undefined when creating a new ProgramIndicator. Should be added as default when creating the model in program-indicator/epics.js

Reply:
Damn. I completely missed this. I'll update ze code.

Reply:
6cfb7cc7f1a8e55154687fb3c283464ff7c52898